osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/SliderBall.cs

```diff
@@ -155,15 +168,35 @@ protected override void Update()
             {
                 // Make sure to use the base version of ReceivePositionalInputAt so that we correctly check the position.
                 Tracking = canCurrentlyTrack
-                           && lastScreenSpaceMousePosition.HasValue
-                           && ReceivePositionalInputAt(lastScreenSpaceMousePosition.Value)
-                           && (drawableSlider?.OsuActionInputManager?.PressedActions.Any(x => x == OsuAction.LeftButton || x == OsuAction.RightButton) ?? false);
+                        && cursorTrackingBall
+                        && (drawableSlider?.OsuActionInputManager?.PressedActions.Any(x => (x == OsuAction.LeftButton || x == OsuAction.RightButton) && trackingActions.Contains(x)) ?? false);
             }
         }
 
         public void UpdateProgress(double completionProgress)
         {
             Position = slider.CurvePositionAt(completionProgress);
         }
+
+        public bool OnPressed(OsuAction action)
+        {
+            if (cursorTrackingBall && (Hit?.Invoke() ?? false))
```

Comment:
you're not checking the action type here. while this works at the moment (the only members of `OsuAction` is `LeftButton` and `RightButton`), but this is not a guarantee. That's why the check is in the `Update` condition.